zerver/webhooks/github/view.py

```diff
@@ -371,15 +372,17 @@ def api_github_webhook(
         request: HttpRequest, user_profile: UserProfile,
         payload: Dict[str, Any]=REQ(argument_type='body'),
         branches: Text=REQ(default=None)) -> HttpResponse:
-    event = get_event(request, payload, branches)
+    event = get_event(request, user_profile, payload, branches)
     if event is not None:
         subject = get_subject_based_on_type(payload, event)
         body = get_body_function_based_on_type(event)(payload)
         check_send_webhook_message(request, user_profile, subject, body)
     return json_success()
 
-def get_event(request: HttpRequest, payload: Dict[str, Any], branches: Text) -> Optional[str]:
-    event = request.META['HTTP_X_GITHUB_EVENT']
+def get_event(request: HttpRequest, user_profile: UserProfile,
+              payload: Dict[str, Any], branches: Text) -> Optional[str]:
+    event = validate_extract_webhook_http_header(
+        'HTTP_X_GITHUB_EVENT', request, user_profile)
```

Comment:
I think a function name and argument structure that's more like this:

`event = extract_header(request, HTTP_X_GITHUB_EVENT)` 

would be cleaner; you can get the user from `request.user`.